### PR TITLE
modify Q48: current solution cannot handle "0 at diagonal" circumstance

### PR DESCRIPTION
**Pull request by peppermin-t**
current solution cannot handle "0 at diagonal" circumstance, since `A[i], A[r] = A[r], A[i]` cannot complete the swap operation indeed.

I add the test instance generating errors given the former solution, and a simpler method I proposed in `solution.py`. I also modify some contents in learn.html accordingly.

### CODE REVIEW

File: Problems/48_rref/solution.py [hunk 1]
```diff
@@ -2,31 +2,46 @@
...
+#                     return A
+#         A[i], A[r] = A[r], A[i]
+#         lv = A[r, lead]
+#         A[r] = A[r] / lv
+#         for i in range(m):
+#             if i != r:
+#                 lv = A[i, lead]
+#                 A[i] = A[i] - lv * A[r]
+#         lead += 1
+#     return A
```
Comment (Open-Deep-ML) [inline Problems/48_rref/solution.py:43]: could you remove the old solution

**Review by Open-Deep-ML — COMMENTED** [review]
I like your solution/fix could you just remove the old solution

**Comment by peppermin-t** [comment]
Sure :) old version already removed.

**Review by moe18 — COMMENTED** [review]
your solution would not deal with this use case np.array([[1,2,3],[2,4,6],[3,6,9]]), the current one will not deal with the use case you provided so a new solution is needed.

**Comment by peppermin-t** [comment]
Solutions considering 0 leadings, and test case containing reducible lines (or sublines) provided.

I think it's also better to mention all these requirements in the question more in details, such as the diagonal 1s can be 0s (or lines may be reducible)

**Comment by Open-Deep-ML** [comment]
I agree, I will update the question description and review this PR this weekend, thank you for the contribution!

**Review by Open-Deep-ML — APPROVED** [review]
looks good

**Pull request merged**